File: PycharmProjects/Python_Class/HW_1/HW_1_5.py
import random

print("Cuanta gente quieres que entre para este obra de teatro? ")
peop = int(input())
print("Cuanto va a costar el ticket?")
cost = float(input())
print("**************************************************************************\n")

table = [["             ", "Edad       ", "Descuento    ", "Dinero descontado"],
         ["Categoria 1: ", "5-14       ", " 35%         $", ""],
         ["Categoria 2: ", "15-19      ", " 25%         $", ""],
         ["Categoria 3: ", "20-45      ", " 10%         $", ""],
         ["Categoria 4: ", "46-65      ", " 25%         $", ""],
         ["Categoria 5: ", "66 y mayor ", " 35%         $", ""]]

guest_list = [0, 0, 0, 0, 0]
disc = [0.35, 0.25, 0.10, 0.25, 0.35]

for _ in range(peop):
    age = random.randint(5, 100)

    if 5 <= age <= 14:
        guest_list[0] += 1
    elif 15 <= age <= 19:
        guest_list[1] += 1
    elif 20 <= age <= 45:
        guest_list[2] += 1
    elif 46 <= age <= 65:
        guest_list[3] += 1
    else:
        guest_list[4] += 1

    for x in range(1, 6):
        hold = guest_list[x-1] * cost * disc[x-1]
        table[x][3] = hold

for x in range(6):
    for y in range(4):
        print(table[x][y], end="")
    print(" ")

File: IdeaProjects/Java_Class/src/HW_1/HW_1_5.java
package HW_1;

import java.text.DecimalFormat;
import java.util.Random;
import java.util.Scanner;

public class HW_1_5 {

    public static void main(String[] args){
        /*
        Un teatro otorga descuentos según la edad del cliente. Determinar la cantidad de dinero que el teatro
        deja de percibir por cada una de las categorías. Tomar en cuenta que los niños menores de 5 años no pueden
        entrar al teatro y que existe un precio único en los asientos. Los descuentos se hacen tomando en cuenta
        el siguiente cuadro:

         */
        Scanner sc = new Scanner(System.in);


        System.out.println("Cuanta gente quieres que entre para este obra de teatro? ");
        int peop = sc.nextInt();
        System.out.println("Cuanto va a costar el ticket?");
        double cost = sc.nextDouble();
        System.out.println("**************************************************************************");
        System.out.println("");


        String[][] table ={ {"             ", "Edad       ", "Descuento    ", "Dinero descontado"},
                            {"Categoria 1: ", "5-14       ", " 35%         $",       ""},
                            {"Categoria 2: ", "15-19      ", " 25%         $",       ""},
                            {"Categoria 3: ", "20-45      ", " 10%         $",       ""},
                            {"Categoria 4: ", "46-65      ", " 25%         $",       ""},
                            {"Categoria 5: ", "66 y mayor ", " 35%         $",       ""}};


        Random ra = new Random();

        int age = 0;

        int[] guestList = new int[5];
        double[] disc = {0.35, 0.25, 0.10, 0.25, 0.35};

        for(int v = 0; v < peop; v++){
            age = ra.nextInt(5, 100);

            if (age >= 5 && age <= 14){
                guestList[0]++;


            } else if (age > 14 && age <= 19){
                guestList[1]++;


            } else if (age > 19 && age <= 45){
                guestList[2]++;


            } else if (age > 45 && age <= 65){
                guestList[3]++;

            } else {
                guestList[4]++;


            }


            double hold = 0;


            DecimalFormat round = new DecimalFormat("#.###");

            for(int x = 1; x <= 5; x++){
            for(int y = 3; y <= 3; y++){
                hold = ((double)guestList[x-1])* cost * disc[x-1] ;
                table[x][y] = Double.toString(Double.parseDouble(round.format(hold)));
            }

            }
        }

        for(int x = 0; x < 6; x++){
            for(int y = 0; y < 4; y++){
                System.out.print(table[x][y]);
            }
            System.out.println(" ");
        }




    }

}


